gnomad_qc/v4/sample_qc/create_sample_qc_metadata_ht_genomes.py

```diff
@@ -0,0 +1,144 @@
+from gnomad_qc.v3.resources.basics import meta as v3_meta
+from gnomad_qc.v4.resources.meta import meta as v4_meta
+from gnomad_qc.v4.resources.meta import meta_tsv_path as v4_meta_tsv_path
+from gnomad_qc.v4.resources.sample_qc import hgdp_tgp_meta_updated
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("create_v4_genomes_sample_meta")
+logger.setLevel(logging.INFO)
+
+
+def import_updated_annotations(ht: hl.Table, subset_ht: hl.Table) -> hl.Table:
+    """
+    Import updated annotations from HGDP/TGP subset meta HT and annotate full meta HT.
+
+    :param ht: v3.1 genomes meta HT
+    :param subset_ht: Updated HGDP + TGP subset meta HT
+    :return: Updated v4.0 genomes meta HT
+    """
+    # Some HGDP/TGP samples in v3.1 meta HT have a prefix of "v3.1::" in the key 's',
+    # but the project_meta.sample_id field does not have this prefix. This is used as
+    # a temp key to join the subset meta HT.
+    release_v3 = ht.filter(ht.release).count()
+    ht1 = ht.filter(ht.subsets.hgdp | ht.subsets.tgp)
+    ht1 = ht1.checkpoint(new_temp_file("genomes_meta", "ht"))
+    ht = ht.filter(~(ht.subsets.hgdp | ht.subsets.tgp))
+    ht = ht.checkpoint(new_temp_file("genomes_meta", "ht"))
+
+    ht1 = ht1.key_by(ht1.project_meta.sample_id)
+    annot = subset_ht[ht1.key]
+    subpop = annot.hgdp_tgp_meta.population
+    freemix = annot.bam_metrics.freemix
+    hard_filters = annot.gnomad_sample_filters.hard_filters
+    hard_filtered = annot.gnomad_sample_filters.hard_filtered
+    release_related = (
+        annot.relatedness_inference.related
+        | annot.gnomad_sample_filters.related_to_nonsubset
+        | annot.gnomad_sample_filters.v4_exome_duplicate
+    )
+    high_quality = annot.gnomad_high_quality
+    release = annot.gnomad_release
+
+    ht1 = ht1.annotate(
+        project_meta=ht1.project_meta.annotate(project_subpop=subpop),
+        bam_metrics=ht1.bam_metrics.annotate(freemix=freemix),
+        sample_filters=ht1.sample_filters.annotate(
+            hard_filters=hard_filters,
+            hard_filtered=hard_filtered,
+            release_related=release_related,
+        ),
+        high_quality=high_quality,
+        release=release,
+    )
+
+    ht1 = ht1.key_by("s").drop("sample_id")
+    ht = ht1.union(ht)
+
+    # Update gnomAD inferred population 'oth' to be 'remaining'.
+    ht = ht.annotate(
+        population_inference=ht.population_inference.annotate(
+            pop=hl.if_else(
+                ht.population_inference.pop == "oth",
+                "remaining",
+                ht.population_inference.pop,
+            )
+        )
+    )
+    release_v4 = ht.filter(ht.release).count()
+    logger.info(
+        "Number of genome release samples will change from %s in v3.1 to %s in v4.0",
+        release_v3,
+        release_v4,
+    )
+    if release_v4 <= release_v3 or release_v4 != release_v3 + 169 - 110:
```

Comment:
just a note that this second part (`release_V3 + 169 - 110`) would need to get updated if we ever had to run this again. what about making this a constant at the top of the script? something like:
```
N_DIFF_SAMPLES = 169 - 110
"""
The number of samples that have been updated between v3.1 and v4.0.
"""
```
and then
```
if release_v4 <= release_v3 or release_v4 != release_v3 + N_DIFF_SAMPLES:
```